src/pages/index.js

```diff
@@ -0,0 +1,335 @@
+      if (openModal) closeModal(openModal);
+    }
+  }
+
+  closeModalButtons.forEach((button) => {
+    const modal = button.closest(".modal");
+    button.addEventListener("click", () => closeModal(modal));
+  });
+
+  if (closeDeleteModalButton) {
+    closeDeleteModalButton.addEventListener("click", () => {
+      console.log("Close (X) button clicked in delete modal!");
+      closeModal(deleteCardModal);
+    });
+  }
+
+  if (cancelDeleteButton) {
+    cancelDeleteButton.addEventListener("click", () => {
+      console.log("Cancel button clicked in delete modal!");
+      closeModal(deleteCardModal);
+    });
+  }
+
+  api
+    .getUserInfo()
+    .then((userData) => {
+      console.log("API Response:", userData);
+      profileName.textContent = userData.name || "Bessie Coleman";
+      profileDescription.textContent = userData.about || "Civil Aviator";
+      profileAvatar.src = userData.avatar.includes("placeholder")
+        ? "./images/avatar.jpg"
+        : userData.avatar;
+    })
+    .catch(console.error);
+  function createCardElement(data) {
+    // ... your existing createCardElement function ...
+  }
+
+  function renderCard(data) {
+    const cardElement = createCardElement(data);
+    cardsContainer.prepend(cardElement);
+  }
+
+  function renderInitialCards(cards) {
+    cards.forEach(renderCard);
+  }
+
+  // Then keep your API call
+  api
+    .getInitialCards()
+    .then((cards) => {
+      console.log("Loaded Cards:", cards);
+      if (!Array.isArray(cards)) {
+        console.error("Error: Expected an array but received:", cards);
+        return;
+      }
+      renderInitialCards(cards);
+    })
+    .catch(console.error);
+
+  function renderInitialCards(cards) {
+    cards.forEach(renderCard);
+  }
+  function renderCard(data) {
+    const cardElement = createCardElement(data);
+    cardsContainer.prepend(cardElement);
+  }
+  api
+    .getInitialCards()
+    .then((cards) => {
+      console.log("Loaded Cards:", cards);
+
+      // Safety check before using cards
+      if (!Array.isArray(cards)) {
+        console.error("Error: Expected an array but received:", cards);
+        return;
+      }
+
+      renderInitialCards(cards); // Now it's defined before calling it!
+    })
+    .catch(console.error);
+
+  function createCardElement(data) {
+    const cardElement = cardTemplate.cloneNode(true).firstElementChild;
+    const cardTitle = cardElement.querySelector(".card__title");
+    const cardImage = cardElement.querySelector(".card__img");
+    const deleteButton = cardElement.querySelector(".card__delete-button");
+    const likeButton = cardElement.querySelector(".card__like-button");
+
+    cardTitle.textContent = data.name;
+    cardImage.src = data.link;
+    cardImage.alt = data.name;
+
+    // ✅ Assign the card ID for delete functionality
+    cardElement.dataset.cardId = data._id;
+    console.log("Assigned card ID:", data._id); // Debugging log
+
```

Comment:
* When I reload the page, all **red** hearts become **transparent** again.

Please, check `isLiked` here